module/VuFind/src/VuFind/Config/YamlReader.php

```diff
@@ -161,10 +180,33 @@ protected function parseYaml($file, $defaultParent = null)
             // Swallow the directive after processing it:
             unset($results['@parent_yaml']);
         }
+        // Check for sections to merge instead of overriding:
+        $mergedSections = [];
+        if (isset($results['@merge_sections'])) {
+            $mergedSections = $results['@merge_sections'];
+            // Swallow the directive after processing it:
+            unset($results['@merge_sections']);
+        }
 
-        // Now load in missing sections from parent, if applicable:
+        // Now load in merged or missing sections from parent, if applicable:
         if (null !== $defaultParent) {
-            foreach ($this->parseYaml($defaultParent) as $section => $contents) {
+            $parentSections = $this->parseYaml($defaultParent);
+            // Process merged sections:
+            foreach ($mergedSections as $path) {
+                $resultElemRef = &$this->getArrayElemRefByPath($results, $path);
+                if (is_array($resultElemRef)) {
```

Comment:
Do we really want to skip logic if the merged path does not exist in the child? We're only going to auto-merge missing top-level sections, so what if we want to merge in a more deeply-nested section from the parent that is not defined in the child? I might be misreading something, but it looks to me like that will not work. Does getArrayElemRefByPath perhaps need a flag to allow it to create missing nodes, and then if the path exists in the parent but not the child, we can build it into the child?

Reply:
You're right, it wasn't working properly for anything but top level. Should be fixed now.